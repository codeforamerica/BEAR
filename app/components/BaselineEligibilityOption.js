import React, { Component } from 'react';
import RadioButton from './RadioButton';

type Props = {
  codeSection: string,
  selectedOption: string
};

export default class BaselineEligibilityOption extends Component<Props> {
  render() {
    const { codeSection, selectedOption } = this.props;
    return (
      <tr>
        <td>
          <p>{`HS ${codeSection} felonies`}</p>
        </td>
        <td>
          <RadioButton
            selected={selectedOption === 'dismiss'}
            value="dismiss"
            group={codeSection}
          />
        </td>
        <td>
          <RadioButton
            selected={selectedOption === 'reduce'}
            value="reduce"
            group={codeSection}
          />
        </td>
      </tr>
    );
  }
}
