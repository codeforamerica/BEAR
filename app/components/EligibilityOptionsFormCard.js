/* eslint-disable react/no-unescaped-entities */
/* eslint-disable no-unused-vars */
// @flow
import React, { Component } from 'react';

import FormCard, {
  FormCardContent,
  FormCardFooter,
  FormCardHeader
} from './FormCard';
import BaselineEligibilityOption from './BaselineEligibilityOption';
import RadioButton from './RadioButton';
import ContinueButton from './ContinueButton';
import GoBackButton from './GoBackButton';

type Props = {
  baselineEligibilityOptions: BaselineEligibilityOptions,
  onEligibilityOptionSelect: (string, string) => void,
  onOptionsConfirm: void => void,
  onOptionsRunScript: void => void,
  onBack: void => void
};

export default class EligibilityOptionsFormCard extends Component<Props> {
  onContinue = () => {
    const { onOptionsConfirm, onOptionsRunScript } = this.props;
    onOptionsConfirm();
    onOptionsRunScript();
  };

  onGoBack = () => {
    const { onBack } = this.props;
    onBack();
  };

  render() {
    const {
      baselineEligibilityOptions,
      onEligibilityOptionSelect
    } = this.props;
    return (
      <FormCard>
        <FormCardHeader>
          Baseline eligibility
          <p className="text--help">
            Choose from the following options to expedite your office's review
            of records from the state summary criminal history information
            database for AB-1793.
          </p>
        </FormCardHeader>
        <FormCardContent>
          <table className="data-table">
            <thead>
              <tr>
                <th>Type of convictions</th>
                <th>Dismiss</th>
                <th>Reduce</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  <p>All HS 11357-60 misdemeanors and infractions</p>
                </td>
                <td>
                  <RadioButton
                    selected={true}
                    value="misdemeanor_infraction"
                    group="no_reduce_option"
                  />
                </td>
                <td />
              </tr>
              {[
                '11357(a)',
                '11357(b)',
                '11357(c)',
                '11357(d)',
                '11358',
                '11359',
                '11360'
              ].map(codeSection => {
                return (
                  <BaselineEligibilityOption
                    key={codeSection}
                    codeSection={codeSection}
                    baselineEligibilityOptions={baselineEligibilityOptions}
                    onEligibilityOptionSelect={onEligibilityOptionSelect}
                  />
                );
              })}
            </tbody>
          </table>
        </FormCardContent>
        <FormCardFooter>
          <ContinueButton onContinue={this.onContinue} />
          <GoBackButton onGoBack={this.onGoBack} />
        </FormCardFooter>
      </FormCard>
    );
  }
}
