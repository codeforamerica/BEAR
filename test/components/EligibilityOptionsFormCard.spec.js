import sinon from 'sinon';
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import EligibilityOptionsFormCard from '../../app/components/EligibilityOptionsFormCard';
import BaselineEligibilityOption from '../../app/components/BaselineEligibilityOption';

Enzyme.configure({ adapter: new Adapter() });
const sandbox = sinon.createSandbox();

function setup() {
  const options = {
    '11357(a)': 'reduce',
    '11357(b)': 'dismiss',
    '11357(c)': 'reduce',
    '11357(d)': 'reduce',
    '11358': 'dismiss',
    '11359': 'reduce',
    '11360': 'dismiss'
  };

  const onOptionsConfirmSpy = sandbox.spy();
  const updateDateSpy = sandbox.spy();
  const onBackSpy = sandbox.spy();
  const component = mount(
    <EligibilityOptionsFormCard
      currentScreen={3}
      baselineEligibilityOptions={options}
      onOptionsConfirm={onOptionsConfirmSpy}
      updateDate={updateDateSpy}
      onBack={onBackSpy}
    />
  );
  return {
    options,
    component,
    onOptionsConfirmSpy,
    updateDateSpy,
    onBackSpy
  };
}
afterEach(() => {
  sandbox.restore();
});

describe('EligibilityOptionsFormCard component', () => {
  it('should render each option with the correct selection (dismiss or reduce)', () => {
    const { component } = setup('');
    expect(
      component.containsAnyMatchingElements([
        <BaselineEligibilityOption codeSection="11357(a)" />
      ])
    ).toEqual(true);
    expect(
      component.containsAnyMatchingElements([
        <BaselineEligibilityOption codeSection="11357(b)" />
      ])
    ).toEqual(true);
    expect(
      component.containsAnyMatchingElements([
        <BaselineEligibilityOption codeSection="11357(c)" />
      ])
    ).toEqual(true);
    expect(
      component.containsAnyMatchingElements([
        <BaselineEligibilityOption codeSection="11357(d)" />
      ])
    ).toEqual(true);
    expect(
      component.containsAnyMatchingElements([
        <BaselineEligibilityOption codeSection="11358" />
      ])
    ).toEqual(true);
    expect(
      component.containsAnyMatchingElements([
        <BaselineEligibilityOption codeSection="11359" />
      ])
    ).toEqual(true);
    expect(
      component.containsAnyMatchingElements([
        <BaselineEligibilityOption codeSection="11360" />
      ])
    ).toEqual(true);
  });

  describe('clicking the continue button', () => {
    it('should call updateDate once', () => {
      const { component, updateDateSpy } = setup('path/to/file');
      component.find('#continue').simulate('click');
      expect(updateDateSpy.called).toBe(true);
      expect(updateDateSpy.callCount).toEqual(1);
    });
    it('should call onOptionsConfirm once', () => {
      const { component, onOptionsConfirmSpy } = setup('path/to/file');
      component.find('#continue').simulate('click');
      expect(onOptionsConfirmSpy.called).toBe(true);
      expect(onOptionsConfirmSpy.callCount).toEqual(1);
    });
  });

  describe('clicking the go back button', () => {
    it('should call onBack once', () => {
      const { component, onBackSpy } = setup('path/to/file');
      component.find('#goback').simulate('click');
      expect(onBackSpy.called).toBe(true);
      expect(onBackSpy.callCount).toEqual(1);
    });
  });

  it('should match exact snapshot', () => {
    const options = { '11357(a)': 'dismiss' };
    const component = (
      <div>
        <EligibilityOptionsFormCard
          currentScreen={3}
          baselineEligibilityOptions={options}
        />
      </div>
    );

    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
