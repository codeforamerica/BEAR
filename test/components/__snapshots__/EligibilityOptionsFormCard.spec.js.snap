// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EligibilityOptionsFormCard component should match exact snapshot 1`] = `
<div>
  <div
    className="form-card"
  >
    <header
      className="form-card__header"
    >
      <h1
        className="form-card__title"
      >
        Baseline eligibility
        <p
          className="text--help"
        >
          Choose whether to dismiss or reduce Prop 64 convictions.
        </p>
      </h1>
    </header>
    <div
      className="form-card__content"
    >
      <table
        className="data-table"
      >
        <thead>
          <tr>
            <th>
              Type of convictions
            </th>
            <th>
              Dismiss
            </th>
            <th>
              Reduce
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <p>
                All Prop 64 misdemeanors and infractions
              </p>
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={true}
                id="misdemeanor_infraction_no_reduce_option"
                name="no_reduce_option"
                onChange={[Function]}
                type="radio"
              />
            </td>
            <td />
          </tr>
          <tr>
            <td>
              <p>
                H&S § 11357 (a) felonies
              </p>
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={true}
                id="dismiss_11357(a)"
                name="11357(a)"
                onChange={[Function]}
                type="radio"
              />
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="reduce_11357(a)"
                name="11357(a)"
                onChange={[Function]}
                type="radio"
              />
            </td>
          </tr>
          <tr>
            <td>
              <p>
                H&S § 11357 (b) felonies
              </p>
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="dismiss_11357(b)"
                name="11357(b)"
                onChange={[Function]}
                type="radio"
              />
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="reduce_11357(b)"
                name="11357(b)"
                onChange={[Function]}
                type="radio"
              />
            </td>
          </tr>
          <tr>
            <td>
              <p>
                H&S § 11357 (c) felonies
              </p>
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="dismiss_11357(c)"
                name="11357(c)"
                onChange={[Function]}
                type="radio"
              />
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="reduce_11357(c)"
                name="11357(c)"
                onChange={[Function]}
                type="radio"
              />
            </td>
          </tr>
          <tr>
            <td>
              <p>
                H&S § 11357 (d) felonies
              </p>
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="dismiss_11357(d)"
                name="11357(d)"
                onChange={[Function]}
                type="radio"
              />
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="reduce_11357(d)"
                name="11357(d)"
                onChange={[Function]}
                type="radio"
              />
            </td>
          </tr>
          <tr>
            <td>
              <p>
                H&S § 11357 felonies (when no sub-section is specified)
              </p>
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="dismiss_11357(no-sub-section)"
                name="11357(no-sub-section)"
                onChange={[Function]}
                type="radio"
              />
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="reduce_11357(no-sub-section)"
                name="11357(no-sub-section)"
                onChange={[Function]}
                type="radio"
              />
            </td>
          </tr>
          <tr>
            <td>
              <p>
                H&S § 11358 felonies including all subsections
              </p>
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="dismiss_11358"
                name="11358"
                onChange={[Function]}
                type="radio"
              />
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="reduce_11358"
                name="11358"
                onChange={[Function]}
                type="radio"
              />
            </td>
          </tr>
          <tr>
            <td>
              <p>
                H&S § 11359 felonies including all subsections
              </p>
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="dismiss_11359"
                name="11359"
                onChange={[Function]}
                type="radio"
              />
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="reduce_11359"
                name="11359"
                onChange={[Function]}
                type="radio"
              />
            </td>
          </tr>
          <tr>
            <td>
              <p>
                H&S § 11360 felonies including all subsections
              </p>
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="dismiss_11360"
                name="11360"
                onChange={[Function]}
                type="radio"
              />
            </td>
            <td>
              <input
                className="radioButton"
                defaultChecked={false}
                id="reduce_11360"
                name="11360"
                onChange={[Function]}
                type="radio"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="form-card__footer"
    >
      <button
        className="button button--primary"
        id="continue"
        onClick={[Function]}
        type="button"
      >
        Continue 
        <i
          className="icon-arrow_forward"
        />
      </button>
      <button
        className="button"
        id="goback"
        type="button"
      >
        Go back 
        <i
          className="icon-replay"
        />
      </button>
    </div>
  </div>
</div>
`;
